Build failure with GCC 15 (defaults to -std=gnu23)

Upcoming GCC 15 defaults to C23 (`-std=gnu23`) which causes setproctitle to fail to build:
```
 x86_64-pc-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -O2 -pipe -march=native -fno-diagnostics-color -DNDEBUG -fPIC -DSPT_VERSION=1.3.4 -DHAVE_SYS_PRCTL_H=1 -I/usr/include/python3.12 -c src/setproctitle.c -o /var/tmp/portage/dev-python/setproctitle-1.3.4-r1/work/setproctitle-1.3.4-python3_12/build/temp.linux-x86_64-cpython-312/src/setproctitle.o
In file included from src/spt_status.h:13,
                 from src/setproctitle.c:16:
src/c.h:18:14: error: two or more data types in declaration specifiers
   18 | typedef char bool;
      |              ^~~~
src/c.h:18:1: warning: useless type name in empty declaration
   18 | typedef char bool;
      | ^~~~~~~
error: command '/usr/bin/x86_64-pc-linux-gnu-gcc' failed with exit code 1
```

This happens because `bool` is a real type and keyword in C23.

Originally reported downstream in Gentoo at https://bugs.gentoo.org/943816.

--- a/HISTORY.rst
+++ b/HISTORY.rst
@@ -5,6 +5,7 @@ Version 1.3.5 (unreleased)
 --------------------------
 
 - Fix bouncing Dock icon on macOS (issue #143).
+- Fix building on C23 compilers (issue #145).
 
 
 Version 1.3.4

--- a/src/c.h
+++ b/src/c.h
@@ -12,20 +12,9 @@
 
 #include "spt_config.h"
 
-#ifndef __cplusplus
-
-#ifndef bool
-typedef char bool;
+#if !defined(__cplusplus) && (!defined(__STDC_VERSION__) || __STDC_VERSION__ <= 201710L)
+#include <stdbool.h>
 #endif
-
-#ifndef true
-#define true	((bool) 1)
-#endif
-
-#ifndef false
-#define false	((bool) 0)
-#endif
-#endif   /* not C++ */
 
 #include <stddef.h>
 

